### PR TITLE
Add "Rarely Used Terms" feature to Ephemera Fields

### LINKED ISSUE #4647: Add a way to select terms which should be grouped as "Rarely Used" for a given project.
In each project there should be a drop-down for each Controlled Vocabulary to select "rarely used" terms. Those should group in their own category at the bottom of the EphemeraFolder ingest.

Newly added terms to a vocabulary should not be marked as "rarely used"

## Implementation

I think we can implement this the same way we implement "favorite terms", which is just store a list of ephemera term IDs in the EphemeraField. See https://github.com/pulibrary/figgy/pull/1961/files for how we implemented that.

### PR DESCRIPTION
**Pull request by tpendragon**
Closes #4647 

Screen Shot 2021-11-03 at 11 51 15 AM

### CODE REVIEW

**Comment by tpendragon** [comment]
Converting to draft to do some refactoring of this load_fields business.

**Comment by tpendragon** [comment]
Ready for review - I think there's more refactoring that could be done in EphemeraFieldDecorator, but I'm unsure of which pattern would make the code actually more readable.

File: app/resources/ephemera_fields/ephemera_field_decorator.rb [hunk 1]
```diff
@@ -21,6 +21,50 @@ def favorite_terms
...
+  # terms at the beginning.
+  def sorted_terms_or_categories
+    return sorted_categories if attribute_name == "subject"
+    non_favorite_terms = vocabulary.terms.select { |x| !favorite_term_ids.include?(x.id) && !rarely_used_term_ids.include?(x.id) }
+    favorite_terms + non_favorite_terms + rarely_used_terms
+  end
+
+  # Returns categories where favorite terms appear at the top and rarely used
+  # terms are removed from categories and placed at the bottom.
+  def sorted_categories
```
Comment (hackartisan) [inline app/resources/ephemera_fields/ephemera_field_decorator.rb:38]: Should we make these private?

File: app/resources/ephemera_fields/ephemera_field_decorator.rb [hunk 2]
```diff
@@ -21,6 +21,50 @@ def favorite_terms
...
+
+  # Returns categories or terms with rarely used terms at the end and favorite
+  # terms at the beginning.
+  def sorted_terms_or_categories
+    return sorted_categories if attribute_name == "subject"
+    non_favorite_terms = vocabulary.terms.select { |x| !favorite_term_ids.include?(x.id) && !rarely_used_term_ids.include?(x.id) }
+    favorite_terms + non_favorite_terms + rarely_used_terms
+  end
+
+  # Returns categories where favorite terms appear at the top and rarely used
```
Comment (hackartisan) [inline app/resources/ephemera_fields/ephemera_field_decorator.rb:36]: So favorite terms appear in both places. It might help to add this to the comment.

**Review by hackartisan — CHANGES_REQUESTED** [review]
I think it looks great. Seems clear and I don't think we need more refactoring. I made a couple of very small comments.

**Comment by tpendragon** [comment]
@hackartisan Thank you! Should be ready for re-review.

**Review by hackartisan — APPROVED** [review]

**Pull request merged**